src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -593,37 +637,73 @@ private static TextSpan GetSpan(Stack<SyntaxNodeOrToken> stack, int first, int l
             return TextSpan.FromBounds(start, end);
         }
 
-        private static Stack<SyntaxNodeOrToken> Combine(Stack<SyntaxNodeOrToken> first, Stack<SyntaxNodeOrToken> next)
+        private static TextSpan GetSpan(Queue<SyntaxNodeOrToken> stack, int first, int length)
         {
-            if (first == null)
+            int start = -1, end = -1, i = 0;
+            foreach (var n in stack)
```

Comment:
it feels like if we're doing this, we might be better suited having just a list that we can index directly into.